django_app/redbox_app/redbox_core/consumers.py

```diff
@@ -200,24 +200,25 @@ def save_ai_message(
         chat_message = ChatMessage(chat=session, text=user_message_text, role=ChatRoleEnum.ai, route=self.route)
         chat_message.save()
         for file, citations in self.citations:
-            file.last_referenced = timezone.now()
-            file.save()
+            if file:
+                file.last_referenced = timezone.now()
+                file.save()
 
             for citation in citations:
                 if file:
                     Citation.objects.create(
                         chat_message=chat_message,
-                        file=file,
                         text=citation.page_content,
+                        source=Citation.Origin.USER_UPLOADED_DOCUMENT,
+                        file=file,
                         page_numbers=parse_page_number(citation.metadata.get("page_number")),
-                        source=Citation.Origin.USER_UPLOADED_DOCUMENT
                     )
                 else:
                     Citation.objects.create(
                         chat_message=chat_message,
-                        url=citation.metadata.get("original_resource_ref"),
                         text=citation.page_content,
-                        source=Citation.Origin(citation.metadata.get("creator_type"))
+                        source=Citation.Origin.WIKIPEDIA,
```

Comment:
yes i know not all external citations will from wikipedia in the future, but this is the case in this PR